core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -1011,6 +1011,20 @@ public TransactionOutput addOutput(TransactionOutput to) {
         return to;
     }
 
+    /**
+     * Replaces an already added output.
+     *
+     * @param index  index of output to replace
+     * @param output output to replace with
+     */
+    public void replaceOutput(int index, TransactionOutput output) {
+        TransactionOutput oldOutput = outputs.remove(index);
+        checkState(oldOutput.isAvailableForSpending());
```

Comment:
Why are you checking this? Maybe we should add a comment explaining why?

Reply:
I thought that if people start replacing outputs in buried in their wallet things will go wrong. Those txns should really be considered immutable.

Reply:
Makes sense. I think you should say something like that in a comment.

Reply:
Will do.

Reply:
I added this: https://github.com/bitcoinj/bitcoinj/compare/2d465f91971cb35818b217647229db29ce22ce7d..9daf18eb0fd5d2b3bf2463f9c24522d11d8c88af